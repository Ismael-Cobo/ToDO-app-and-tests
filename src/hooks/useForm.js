import { useState } from "react"


export const useForm = ( initialState = {} ) => {
    
    const [values, setValues] = useState(initialState);

    const reset = () => {
        setValues( initialState );
    }


    const handleInputChange = ({ target }) => {


        // el target contiene las propiedades del input, es decir,
        // el name, value, class, prototypes, etc...
        // por eso ponemos target.name y target.value
        // ej:
        // Queremos cambiar el value del input name a ismael 2
        // name: 'ismael 2'

        setValues({
            ...values,
            [ target.name ]: target.value
        });

    }

    return [ values, handleInputChange, reset ];

}
