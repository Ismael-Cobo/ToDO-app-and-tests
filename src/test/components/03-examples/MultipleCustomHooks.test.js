import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import { useFetch } from "../../../hooks/useFetch"
jest.mock("../../../hooks/useFetch")

describe('prueba del componente <MultipleCustomHooks />', () => {

    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper).toMatchSnapshot()


    })

    test('debe de mostrar la informaciÃ³n', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Ismael',
                quote: 'Hey '
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper).toMatchSnapshot()

    })


})