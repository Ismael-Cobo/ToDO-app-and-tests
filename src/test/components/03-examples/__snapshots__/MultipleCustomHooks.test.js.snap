// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba del componente <MultipleCustomHooks /> debe de mostrar la informaci√≥n 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    >
       
      Hey 
       
    </p>
    <footer
      className="blockquote-footer"
    >
       
      Ismael
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;

exports[`prueba del componente <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
